pkg-r/R/client_state.R

```diff
@@ -87,8 +87,36 @@ method(client_set_ui, S7::new_S3_class(c("Chat", "R6"))) <-
     # TODO-barret-future; In shinychat, make this a single/internal custom message call to send all the messages at once (and then scroll)
 
     msgs <- contents_shinychat(client)
-    lapply(msgs, function(x) {
-      chat_append(id, x$content, role = x$role)
+    lapply(msgs, function(msg_turn) {
+      is_list <- is.list(msg_turn$content) &&
+        !inherits(msg_turn$content, c("shiny.tag", "shiny.taglist"))
+
+      if (is_list) {
+        chat_append_message(
+          id,
+          msg = list(role = msg_turn$role, content = ""),
+          operation = "append",
+          chunk = "start",
+        )
+
+        for (content_part in msg_turn$content) {
+          chat_append_message(
+            id,
+            msg = list(role = msg_turn$role, content = content_part),
+            operation = "append",
+            chunk = TRUE,
+          )
+        }
+
+        chat_append_message(
+          id,
+          msg = list(role = msg_turn$role, content = ""),
+          operation = "append",
+          chunk = "end",
+        )
```

Comment:
```suggestion
        stream <- coro::async_generator(function() {
          for (x in msg_turn$content) coro::yield(x)
        })
        chat_append(id, stream, msg_turn$role)
})
```

Reply:
oh that's nice! thanks!

Reply:
Actually I think it needs to be a sync generator, but still this is nice